devices/blepsynth/blepsynth.cc

```diff
@@ -987,6 +968,64 @@ class BlepSynth : public AudioProcessor {
     need_update_volume_envelope_ = false;
     need_update_filter_envelope_ = false;
   }
+  void
+  render (uint n_frames) override
+  {
+    assert_return (n_ochannels (stereout_) == 2);
+
+    float *left_out = oblock (stereout_, 0);
+    float *right_out = oblock (stereout_, 1);
+
+    floatfill (left_out, 0.f, n_frames);
+    floatfill (right_out, 0.f, n_frames);
+
+    uint offset = 0;
+    MidiEventInput evinput = midi_event_input();
+    for (const auto &ev : evinput)
+      {
+        assert_return (ev.frame >= 0); // TODO: should be unsigned anyway, issue #26
+
+        // ensure that new offset from event is not larger than n_frames
+        uint new_offset = ev.frame;
+        if (new_offset > n_frames)
+          {
+            printerr ("*** BlepSynth: event offset is after end of block (new_offset = %d, n_frrames = %d)\n", new_offset, n_frames);
+            new_offset = n_frames;
+          }
+        // ensure that new offset is after last event
+        if (new_offset < offset)
```

Comment:
This has been fixed recently, and in any case, checking this in every processor in the inner event loop is really not the way to go. Please remove unqualified (likely/unlikely) extraneous debugging code.

Reply:
Done (7f8062a86021ff89d9465fd780df30f12d5624fc). Note that now if the ordering or in-range guarantee is violated, the code will crash.

But I tested after rebasing this to trunk and everything works as exected.